libbeat/cmd/instance/metrics.go

```diff
@@ -36,3 +55,38 @@ func reportInfo(_ monitoring.Mode, V monitoring.Visitor) {
 	uptime := int64(delta.Seconds())*1000 + int64(delta.Nanoseconds()/1000000)
 	monitoring.ReportInt(V, "uptime.ms", uptime)
 }
+
+func reportCPU(_ monitoring.Mode, V monitoring.Visitor) {
+	V.OnRegistryStart()
+	defer V.OnRegistryFinished()
+
+	cpuUsage, normalizedCPU := getCPUUsage()
+	monitoring.ReportFloat(V, "usage", cpuUsage)
+	monitoring.ReportFloat(V, "usage.normalized", normalizedCPU)
+}
+
+func getCPUUsage() (float64, float64) {
+	pid := os.Getpid()
+
+	sample := struct {
+		time      time.Time
+		procTimes sigar.ProcTime
+	}{
+		time.Now(),
+		sigar.ProcTime{},
+	}
```

Comment:
We are using the same anonymized struct in [two different places in the code](https://github.com/elastic/beats/pull/5545/files#diff-b248fa29eddb03765933dc1227905d4bR13) We can make a concrete struct maybe called `CpuSample` and reuse it or we could also encapsulate the creation of the sample into his own method and return the sample or error.